logstash-core/src/main/java/org/logstash/secret/cli/SecretStoreCli.java

```diff
@@ -0,0 +1,154 @@
+                            return;
+                        }
+                        add(secretStore, id, SecretStoreUtil.asciiCharToBytes(secret));
+                    } else {
+                        SecretStoreUtil.clearBytes(s);
+                        terminal.write(String.format("%s already exists. Overwrite ? [y/N] ", secretId));
+                        if (isYes(terminal.readLine())) {
+                            terminal.write(String.format("Enter value for %s: ", secretId));
+                            char[] secret = terminal.readSecret();
+                            add(secretStore, id, SecretStoreUtil.asciiCharToBytes(secret));
+                        }
+                    }
+                } else {
+                    terminal.writeLine(String.format("ERROR: Logstash keystore not found. Use 'create' command to create one."));
+                }
+                break;
+            }
+            case REMOVE: {
+                if (secretId == null || secretId.isEmpty()) {
+                    terminal.writeLine("ERROR: You must supply a value to add.");
+                    return;
+                }
+                SecretIdentifier id = new SecretIdentifier(secretId);
+
+                SecretStore secretStore = SecretStoreFactory.load(config);
+                byte[] s = secretStore.retrieveSecret(id);
+                if (s == null) {
+                    terminal.writeLine(String.format("ERROR: '%s' does not exist in the Logstash keystore.", secretId));
+                } else {
+                    SecretStoreUtil.clearBytes(s);
+                    secretStore.purgeSecret(id);
+                    terminal.writeLine(String.format("Removed '%s' from the Logstash keystore.", id.getKey()));
+                }
+                break;
+            }
+            case HELP: {
+                terminal.writeLine("");
+                terminal.writeLine("Commands");
+                terminal.writeLine("--------");
+                terminal.writeLine("create - Creates a new Logstash keystore");
+                terminal.writeLine("list   - List entries in the keystore");
+                terminal.writeLine("add    - Add a value to the keystore");
+                terminal.writeLine("remove - Remove a value from the keystore");
+                terminal.writeLine("");
+                break;
+            }
+        }
+    }
+
+    private void add(SecretStore secretStore, SecretIdentifier id, byte[] secret) {
+        secretStore.persistSecret(id, secret);
+        terminal.writeLine(String.format("Added '%s' to the Logstash keystore.", id.getKey()));
+        SecretStoreUtil.clearBytes(secret);
+    }
+
+    private void create(SecureConfig config) {
+        if (System.getenv(SecretStoreFactory.ENVIRONMENT_PASS_KEY) == null) {
+            terminal.write(String.format("WARNING: The keystore password is not set. Please set the environment variable `%s`. Failure to do so will result in" +
```

Comment:
> WARNING: The keystore password is not set. Please set the environment variable `LOGSTASH_KEYSTORE_PASS`. Failure to do so will result in reduced security. Continue anyway ? [y/N]

It's unclear what behavior the user should expect by pressing `y` here. Maybe "Continue without password protection on the keystore?"

Reply:
> Continue without password protection on the keystore?

+ 1 , will update

I struggled a bit on the wording of this.